scim2-sdk-common/src/main/java/com/unboundid/scim2/common/GenericScimResource.java

```diff
@@ -585,6 +631,24 @@ public GenericScimResource addStringValues(final String path,
     return addStringValues(Path.fromString(path), values);
   }
 
+  /**
+   * Alternate version of {@link #addStringValues(String, List)}.
+   *
+   * @param path    The path to the attribute that should be updated.
+   * @param value1  The first value. This must not be {@code null}.
+   * @param values  An optional field for additional values. Any {@code null}
+   *                values will be ignored.
+   * @return        This object.
+   *
+   * @throws ScimException  If the path is invalid.
+   */
+  public GenericScimResource addStringValues(final String path,
```

Comment:
One thing I noticed is that this file (and `PatchOperation`) have methods for `addBooleanValues()`. Plural. This seems wrong to me. I can't think of a situation where it makes sense to have a multi-valued boolean attribute. At most, it'd ever be `[ true, false ]` (if there are no repeating values), and that array itself doesn't make sense to me.

I can consider adding a `addBooleanValue(String path, boolean value)` method and deprecating the old one, but I thought I'd point this out first.

Reply:
Yes, that's odd. It does look like `replaceValue(String path, Boolean value)` handles the case of adding or updating a single-valued boolean property, though.

Reply:
Thanks, I think I'll move forward with marking this method as deprecated, then. Do you think it would be worth adding a `addBooleanValue` method that just calls `replaceValue`? It may not be obvious to clients that this is the appropriate method when they are looking to add a value.